design-proposals/descheduler-support.md

```diff
@@ -0,0 +1,106 @@
+
+### Pod Eviction Webhook
+There is a [validating webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) that intercepts all eviction requests created in the cluster.
+The webhook approves all eviction requests for pods that are not `virt-launcher` pods.
+
+The webhook takes action, based on the VMI's eviction strategy (defined on the VMI with a fallback to a cluster-wide definition):
+
+| Migration Strategy    | VMI is migrateable                    | VMI is not migrateable | Remarks                               |
+|-----------------------|---------------------------------------|------------------------|---------------------------------------|
+| None                  | N/A                                   | N/A                    | Always approve                        |
+| LiveMigrate           | Mark the VMI for eviction and approve | Deny                   |                                       |
+| LiveMigrateIfPossible | Mark the VMI for eviction and approve | Approve                |                                       |
+| External              | N/A                                   | N/A                    | Mark the VMI for eviction and approve |
+
+
+### Disruption Budget Controller
+`virt-launcher` pods have another layer of protection against evictions using [PDBs](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The controller is responsible for managing PDBs:
+
+The controller decides whether a VMI needs protection against eviction based on its migration strategy (defined on the VMI with a fallback to a cluster-wide definition):
+
+| Migration Strategy    | Requires a PDB                 | 
+|-----------------------|--------------------------------|
+| None                  | False                          |
+| LiveMigrate           | True                           |
+| LiveMigrateIfPossible | Only if the VMI is migrateable |
+| External              | True                           |
+
+### Evacuation Controller
+The controller looks for potential VMIs to evict and tries to migrate them to another node.
+
+### Migration Controller
+The controller manages the migration process initiated by the Evacuation controller.
+The controller migrates the `virt-launcher` pod from the source node to a target node.
+After the migration process is completed, the original `virt-launcher` pod is deleted from the source node and the eviction process ends.
+
+## Motivation
+Cluster administrators want their workloads to be balanced across the cluster.
+These workloads include KubeVirt VMs.
+
+## Goals
+1. Enable KubeVirt to work on existing K8s clusters that already have a descheduler deployed.
+2. Enable cluster admins to introduce a descheduler to an existing cluster with KubeVirt deployed.
+
+## Non Goals
+The descheduler should respect the two annotations included in this document.
```

Comment:
An `annotation` sounds like an implementation detail which is not clear at this stage.
Perhaps you should describe it in a different way. E.g. existing desch... to be aware of VM workloads.

Reply:
Rephrased it.